genie-web/src/test/groovy/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobTerminationServiceImplSpec.groovy

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.rpc.grpc.services.impl.v4
+
+import com.netflix.genie.common.dto.JobStatus
+import com.netflix.genie.common.exceptions.GenieException
+import com.netflix.genie.proto.*
+import com.netflix.genie.web.services.JobSearchService
+import io.grpc.stub.StreamObserver
+import spock.lang.Specification
+
+
+/**
+ * Specifications for the {@link GRpcJobTerminationServiceImpl} class.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+class GRpcJobTerminationServiceImplSpec extends Specification {
+
+    GRpcJobTerminationServiceImpl service
+    String jobId
+    JobKillListenerRequest request
+    JobKillListenerResponse response
+    JobSearchService jobSearchService = Mock()
+    StreamObserver<SyncResponse> responseObserver = Mock()
+
+    void setup() {
+        this.jobId = UUID.randomUUID().toString()
+        this.request = JobKillListenerRequest.newBuilder().setJobId(this.jobId).build()
+        this.response = JobKillListenerResponse.newBuilder().setJobId(this.jobId).build()
+        this.service = new GRpcJobTerminationServiceImpl(this.jobSearchService)
+    }
+
+    def "Can register for kill notification"() {
+
+        when:
+        service.registerForKillNotification(request, responseObserver)
+
+        then:
+        service.activeStreams.get(jobId) != null
+    }
+
+    def "Can kill unfinished jobs and clean up response observer"() {
```

Comment:
I think this test covers a superset of the lines covered in the previous one (i.e. the previous does not add anything)